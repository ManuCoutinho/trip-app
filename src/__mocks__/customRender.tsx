import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ReactElement, FC, ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'
import theme from '../styles/theme/theme'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
