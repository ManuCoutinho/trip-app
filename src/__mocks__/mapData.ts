import { MapProps } from 'components/Map/types'

export default {
  places: [
    {
      id: 'first-place',
      name: 'Rio de Janeiro',
      slug: 'rio-de-janeiro',
      location: {
        latitude: -22.925453,
        longitude: -43.277763
      }
    },
    {
      id: 'second-place',
      name: 'Lagos',
      slug: 'lagos',
      location: {
        latitude: 6.460649,
        longitude: 3.333309
      }
    }
  ]
} as MapProps
