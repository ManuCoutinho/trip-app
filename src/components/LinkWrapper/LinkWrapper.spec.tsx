import { screen } from '@testing-library/react'
import { render } from '../../__mocks__/customRender'
import { LinkWrapper } from '.'

const linkProps = {
  children: 'children',
  href: 'about',
  title: 'about'
}
describe('<LinkWrapper/>', () => {
  it('should render a link and children correctly ', () => {
    render(<LinkWrapper {...linkProps} />)
    const link = screen.getByRole('link', { name: /about/i })
    expect(link).toBeInTheDocument()
    expect(link).toHaveAttribute('href', '/about')
    expect(link.firstChild).toHaveTextContent(linkProps.children)
  })
  it('should match to snapshot ', () => {
    const { container } = render(<LinkWrapper {...linkProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
