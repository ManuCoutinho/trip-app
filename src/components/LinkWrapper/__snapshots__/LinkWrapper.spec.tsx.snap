// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkWrapper/> should match to snapshot  1`] = `
<div
  class="sc-bcXHqe fqhbAx"
>
  <a
    href="/about"
    title="link to about"
  >
    children
  </a>
</div>
`;
